yt_dlp/extractor/niconico.py

```diff
@@ -934,8 +934,6 @@ class NiconicoLiveIE(InfoExtractor):
     _KNOWN_LATENCY = ('high', 'low')
 
     def _real_extract(self, url):
-        if not websockets:
-            raise ExtractorError('websockets library is not available. Please install it.', expected=True)
```

Comment:
We want to error early

Reply:
I'll add an equivalent error in ydl.urlopen() when the unsupported request errors come back.

Reply:
It will then still error only after the initial webpage request. Checking dependency and erroring before any request is better. We do the same for cryptodome deps

Reply:
That's fine though, that is the intended behaviour.

Reply:
I'll add this to the "nice to have" list that we can work on later. It's not required for this to be merged imo. It requires a bit of work which is being started in #7595